torch/testing/_internal/common_utils.py

```diff
@@ -1324,11 +1324,10 @@ def assertWarnsOnceRegex(self, category, regex=''):
                 torch.set_warn_always(prev)
                 if len(ws) == 0:
                     self.fail('no warning caught')
-                if len(ws) > 1:
-                    self.fail('too many warnings caught: %s' % '\n    '.join([str(w) for w in ws]))
-                self.assertTrue(type(ws[0].message) is category)
-                self.assertTrue(re.match(pattern, str(ws[0].message)),
-                                f'{pattern}, {ws[0].message}')
+                for w in ws:
+                    self.assertTrue(type(w.message) is category)
+                    self.assertTrue(re.match(pattern, str(w.message)),
+                                    f'{pattern}, {w.message}')
```

Comment:
This condition was too strict: it is OK if there is more than one warning as long as they all match the same category and regex.

Reply:
SGTM!

Reply:
There was a failing test for **not** allowing more than one warning, in 04b9ec93b3 I redid it to test that the wrapper **does** allow more than one warning